### ISSUE TITLE
Program just hangs here forever

### ISSUE THREAD

**Issue by privacyguy123** [Meetem/RazerBladeSharp#1]
![image](https://user-images.githubusercontent.com/50337995/151715990-ad3d12ec-1190-430e-9cda-dc3b52573ed2.png)

Not a clue how to go about debugging this ... but I would love to be able to get rid of Synapse 3 bloat by control the fan some other way.

**Comment by Meetem** [comment]
So you are stuck on requesting status?

**Comment by Meetem** [comment]
I'am not sure about Blade 2019 Pro, cause never worked with that laptop, may be we just don't need to request status here.

**Comment by privacyguy123** [comment]
You would know better than  me ... like I say, this stuff is way over my head just looking for a way to get rid of Synapse 3 bloat, I don't even care about RGB it's just the fan controls that I need - not even Notebook Fan Control works. :/ 

Worth noting the fan is 2300-4300 RPM rather than what that CMD output states - I assume that's RPM?

**Comment by Meetem** [comment]
Then you can safely comment out status query

**Comment by privacyguy123** [comment]
Now I get this:

![image](https://user-images.githubusercontent.com/50337995/151807099-04b828cc-4017-42cc-a156-0eafca5fa58e.png)

Ain't changing any options though

**Comment by Meetem** [comment]
Oh… Looks like Blade 2019 Pro using some different packets :c can you
please use Zaddig tool and change your laptop driver to WinUSB (it’s on
HIDUSB by default), then try again

**Comment by privacyguy123** [comment]
I have 3 of these entries - do they all need changed?

![image](https://user-images.githubusercontent.com/50337995/151860431-bf63fe6c-c059-4b0e-acbf-3b4a8b2f1362.png)

**Comment by Meetem** [comment]
Right one (where WinUSB is written) is replace variant, since it's already
selected, you just need to press Replace Driver (it can take up to 1 minute)

пн, 31 янв. 2022 г. в 22:35, privacyguy123 ***@***.***>:

**Comment by privacyguy123** [comment]
Interface 2 seems to work for me - the other ones break keyboard functionality. Will do some more testing. :)

**Comment by Meetem** [comment]
Glad to hear it! Do status request works too?

пн, 31 янв. 2022 г. в 23:34, privacyguy123 ***@***.***>:

**Comment by privacyguy123** [comment]
RGB seems busted completely unfortunately - what do I write to set all keys to the same color? :S

It seems "numRetries" should be at 2 to make sure the fans speed up all the way - sometimes they stop before getting to full speed?

A lot of people would love this little tool over on the Razer forums if it had some sort of easier GUI!

**Comment by Meetem** [comment]
actually setting correct power mode after fan speed is required, so if you
need to set fan speed, follow it with set power mode command. And disable
razer services (along with completely closing synapse applications) cause
it can interfere with device. You can pass numRetries more than 1
otherwise.

For setting flat color just fill up KeyboardRow with the same color along
all keys. In the example only single key is filled, you just need to add
for()


вт, 1 февр. 2022 г. в 14:40, privacyguy123 ***@***.***>:

**Comment by privacyguy123** [comment]
Sorry I am not a coder - I get an error if I change this line to for()

![image](https://user-images.githubusercontent.com/50337995/151963193-e95b68c5-6ae5-4213-9dcc-73e7f54918ce.png)

**Comment by Meetem** [comment]
Oh, ok. Then change it to:
```
for(int i = 0; i < 6; i++)
{
  var row = new KeyboardRow((byte)i);
  for(int j = 0; j < 15; j++)
    row.keys[j] = new Rgb24(255,255,255);

  _laptop.SendKeyboardRow(row);
}
```

**Comment by privacyguy123** [comment]
Thank you Sir! A lot of people on the Razer forums will love this.

**Comment by privacyguy123** [comment]
Is it possible there is a bug with Razer Blade Pro 17 2019 as it has 2 sets of fans? There is 4 fans in this machine altogether - I seem to have to run the program twice to get them to go full speed.

**Comment by Meetem** [comment]
Yes, there might be a bug, I have only tested 2019 15” advanced, I don’t
have other models :c

**Comment by privacyguy123** [comment]
Is there some info dump I can send you from this machine so we can control both sets of fans? :) 

I think your program is sending information to them randomly.

**Comment by Meetem** [comment]
You can try to repeat set fan command twice

**Comment by privacyguy123** [comment]
Should the numbers correspond to the RPM? This machine is 2300-4300 rather than 3100-5300

**Comment by Meetem** [comment]
Yes, it should, I haven’t real info so just put default which is 3100-5300

**Comment by privacyguy123** [comment]
I am a little confused as the 2nd set of fans shouldn't even be on if power mode is Balanced - so there must be something else wrong under the hood ...

**Comment by privacyguy123** [comment]
@Meetem is there any way I can send you info from this laptop so we can alter the code to make sure it supports the fans? Every time I run the example the fans go a different speed with the same values ...

**Comment by privacyguy123** [comment]
@Meetem is it possible to get this progam to send Chroma info to mouse and keyboard also?

**Comment by Meetem** [comment]
> @Meetem is it possible to get this progam to send Chroma info to mouse and keyboard also?

Sorry for long delay, we have a situation here. I think there is software for doing so, cause Razer Blade and mouse/kb protocols much different (and more popular)

**Comment by Meetem** [comment]
@privacyguy123 hey, I've updated my Simple.exe, you can try now separate fan speeds like this (in command line):
Simple.exe 5000 0 5000
First 5000 is first fan speed,
0 is power mode (balanced)
Second 5000 is 2nd fan speed.
You can set fan speed to 0 or auto.
Also you can now pass keybright=100 (100% keyboard brightness) after first three argument, and keycolor=255,0,255 for keyboard color.

If status query doesn't work for you for some reason, also add disable_get_description

**Comment by Meetem** [comment]
Also note that, for some reason on my laptop power mode=1 (custom) with 5300 fan speed doesn't actually set maximal fan speed, which can also apply to yours. For balanced mode 5300 works just like is should, setting the max speed.
